File: Data_Structures/Tree/Binary_Tree_Traversal/Binary_Tree_Traversal_Recursive.java
public class Binary_Tree_Traversal_Recursive {

	static class Node {
		int data;
		int depth;
		Node left, right;

		Node(int data) {
			this.data = data;
		}
	}

	public Node insert(Node root, int value) {
		if (root == null) {
			return new Node(value);

		}
		if (root.data > value) {
			root.left = insert(root.left, value);
		} else {
			root.right = insert(root.right, value);
		}
		return root;
	}

//PreOrder Traversals	
	public void PreOrder(Node root) {
		if (root == null)
			return;
		System.out.print(root.data + " ");
		PreOrder(root.left);
		PreOrder(root.right);

	}

//InOrder Traversal
	public void InOrder(Node root) {
		if (root == null)
			return;

		InOrder(root.left);
		System.out.print(root.data + " ");
		InOrder(root.right);
	}

//PostOrder Traversal
	public void PostOrder(Node root) {
		if (root == null)
			return;

		PostOrder(root.left);
		PostOrder(root.right);
		System.out.print(root.data + " ");
	}

//Driver Code	
	public static void main(String[] args) {
		Binary_Tree_Traversal_Recursive binarytree= new Binary_Tree_Traversal_Recursive();
		Node root = NULL;
		  root = binarytree.insert(root, 8);
		  root = binarytree.insert(root, 3);
		  root = binarytree.insert(root, 1);
		  root = binarytree.insert(root, 6);
		  root = binarytree.insert(root, 7);
		  root = binarytree.insert(root, 10);
		  root = binarytree.insert(root, 14);
		  root = binarytree.insert(root, 4);
		  
		  System.out.println("Inorder traversal: ");
		  inorder(root);
		  System.out.println("Postorder traversal: ");
		  Postorder(root);
		  System.out.println("Preorder traversal: ");
		  Preorder(root);
		
	}
}

File: Data_Structures/Tree/Binary_Tree_Traversal/BinaryTree_Traversal_Recursive.cs
// C# program for different tree traversals 
using System;

/* Class containing left and right child of current node and key value*/

class Node {
  public int key;
  public Node left,
  right;
  public Node(int item) {
    key = item;
    left = right = null;
  }
}

class BinaryTree {
  Node root;
  BinaryTree() {
    root = null;
  }
  /* Given a binary tree, print its nodes according to the "bottom-up" postorder traversal. */
  void printPostorder(Node node) {
    if (node == null) return;
    // first recur on left subtree 
    printPostorder(node.left);
    // then recur on right subtree 
    printPostorder(node.right);
    // now deal with the node 
    Console.Write(node.key + " ");
  }
  /* Given a binary tree, print 
its nodes in inorder*/
  void printInorder(Node node) {
    if (node == null) return;
    /* first recur on left child */
    printInorder(node.left);
    /* then print the data of node */
    Console.Write(node.key + " ");
    /* now recur on right child */
    printInorder(node.right);
  }
  /* Given a binary tree, print 
its nodes in preorder*/
  void printPreorder(Node node) {
    if (node == null) return;
    /* first print data of node */
    Console.Write(node.key + " ");
    /* then recur on left sutree */
    printPreorder(node.left);
    /* now recur on right subtree */
    printPreorder(node.right);
  }
  // Wrappers over above recursive functions 
  void printPostorder() {
    printPostorder(root);
  }
  void printInorder() {
    printInorder(root);
  }
  void printPreorder() {
    printPreorder(root);
  }

  // Driver Code 
  static public void Main(String[] args) {
    BinaryTree tree = new BinaryTree();
    tree.root = new Node(1);
    tree.root.left = new Node(2);
    tree.root.right = new Node(3);
    tree.root.left.left = new Node(4);
    tree.root.left.right = new Node(5);
    Console.WriteLine("Preorder traversal " + "of binary tree is ");
    tree.printPreorder();
    Console.WriteLine("\nInorder traversal " + "of binary tree is ");
    tree.printInorder();
    Console.WriteLine("\nPostorder traversal " + "of binary tree is ");
    tree.printPostorder();
  }
}